Mage.Client/src/main/java/mage/client/components/ext/dlg/DialogContainer.java

```diff
@@ -29,63 +32,23 @@ public class DialogContainer extends JPanel {
     private boolean drawContainer = true;
     private final DialogManager.MTGDialogs dialogType;
 
+    public void setAlpha(int alpha) {
+        this.alpha = alpha;
+    }
+
     public DialogContainer(DialogManager.MTGDialogs dialogType, DlgParams params) {
         setOpaque(false);
         this.dialogType = dialogType;
 
         setLayout(null);
         drawContainer = true;
 
-        switch (dialogType) {
-
-            case MESSAGE: {
-                if (params.type == MessageDialogType.WARNING) {
-                    backgroundColor = new Color(255, 0, 0, 90);
-                } else {
-                    backgroundColor = new Color(0, 0, 0, 90);
-                }
-                alpha = 0;
-                break;
-            }
-
-            case CHOICE: {
-                backgroundColor = new Color(0, 0, 0, 110);
-                alpha = 0;
-                ChoiceDialog dlg = new ChoiceDialog(params, "Choose");
-                add(dlg);
-                dlg.setLocation(X_OFFSET + 10, Y_OFFSET + 10);
-                dlg.updateSize(params.rect.width - 80, params.rect.height - 80);
-                break;
-            }
-
-            case GRAVEYARD: {
-                backgroundColor = new Color(0, 0, 0, 110);
-                alpha = 0;
-                ChoiceDialog dlg = new ChoiceDialog(params, "Graveyard");
-                add(dlg);
-                dlg.setLocation(X_OFFSET + 10, Y_OFFSET + 10);
-                dlg.updateSize(params.rect.width - 80, params.rect.height - 80);
-                break;
-            }
-
-            case EXILE: {
-                backgroundColor = new Color(250, 250, 250, 50);
-                alpha = 0;
-                ChoiceDialog dlg = new ChoiceDialog(params, "Exile");
-                add(dlg);
-                dlg.setLocation(X_OFFSET + 10, Y_OFFSET + 10);
-                dlg.updateSize(params.rect.width - 80, params.rect.height - 80);
-                break;
-            }
-
-            case EMBLEMS: {
-                backgroundColor = new Color(0, 0, 50, 110);
-                alpha = 0;
-                ChoiceDialog dlg = new ChoiceDialog(params, "Command Zone (Commander, Emblems and Planes)");
-                add(dlg);
-                dlg.setLocation(X_OFFSET + 10, Y_OFFSET + 10);
-                dlg.updateSize(params.rect.width - 80, params.rect.height - 80);
-                break;
+        if(dialogType.getDialogClass() != null) {
+            try {
+                MTGDialog dialog = dialogType.getDialogClass().getDeclaredConstructor().newInstance();
+                backgroundColor = dialog.initialize(this, params, X_OFFSET, Y_OFFSET);
+            } catch(NoSuchMethodException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
+//                logger.error("Error while creating DialogClass instance.");
```

Comment:
Why error ignored here? Wrong dialog class is a non standard use case, so app must inform about critical error.